### ISSUE TITLE
Code works with IB demo account but not with paper trading account.

### ISSUE THREAD

**Issue by rastegar527** [ranaroussi/qtpylib#47]
I run this program with IB demo account and I am able to get data, however it doesn't work with a paper trading account that I have received from IB. I receive no data! I have contacted IB and they have given me the right to receive the stock data.

```python
from qtpylib.algo import Also
from qtpylib.indicators import macd#, air
import pandas as pd
from datetime import date time
import datetime

class TestStrategy(Algo):
    def on_start(self):
         pass
    def on_tick(self, instrument):
        pass
    def on_bar(self, instrument):      
        lookback = 28
        for symbol in self.instruments:
            instr = self.get_instrument(symbol)
            bars = instr.get_bars(lookback = loopback)
            print(symbol, len(bars))
        
if __name__ == "__main__":
    strategy = TestStrategy(
        #instruments = prescreening('2014-04-25', '2014-04-29'),
        instruments = ['IBKR', 'AAPL', 'OXY', 'SLB'], 
        resolution  = "1T" # 1Min bar resolution (Pandas "resample" resolutions)
    )
    strategy.run()

```

**Comment by ranaroussi** [comment]
Closing this issue as it's related to issue #46 - will post solution there

**Comment by rastegar527** [comment]
This is a different issue than #46. After the new update the look back problem is resolved when I connect to demo IB account. However, when I connect to IB paper trading account, I receive no data! absolutely zero. Any suggestion? Thanks Ran!

**Comment by ranaroussi** [comment]
Reopening...

Not sure why this happens. I assume you have the relevant market data subscription on your paper account?

**Comment by rastegar527** [comment]
I have called IB and they say I have "professional US Securities Lite Bundle Market Data".

**Comment by rastegar527** [comment]
I used this piece of code and it seems I can receive the tick data from my paper trading account using IbPy library.  So that tells me it shouldn't be a data subscription issue:

```
from ib.ext.Contract import Contract
from ib.opt import ibConnection, message
from time import sleep

# print all messages from TWS
def watcher(msg):
    print (msg)

# show Bid and Ask quotes
def my_BidAsk(msg):
    if msg.field == 1:
        print ('%s:%s: bid: %s' % (contractTuple[0],
                   contractTuple[6], msg.price))
    elif msg.field == 2:
        print ('%s:%s: ask: %s' % (contractTuple[0], contractTuple[6], msg.price))

def makeStkContract(contractTuple):
    newContract = Contract()
    newContract.m_symbol = contractTuple[0]
    newContract.m_secType = contractTuple[1]
    newContract.m_exchange = contractTuple[2]
    newContract.m_currency = contractTuple[3]
    newContract.m_expiry = contractTuple[4]
    newContract.m_strike = contractTuple[5]
    newContract.m_right = contractTuple[6]
    print ('Contract Values:%s,%s,%s,%s,%s,%s,%s:' % contractTuple)
    return newContract

if __name__ == '__main__':
    con = ibConnection(port = 4001, clientId = 123)
    con.registerAll(watcher)
    showBidAskOnly = False  # set False to see the raw messages
    if showBidAskOnly:
        con.unregister(watcher, message.tickSize, message.tickPrice,
                       message.tickString, message.tickOptionComputation)
        con.register(my_BidAsk, message.tickPrice)
    con.connect()
    sleep(1)
    tickId = 59
    
    # Note: Option quotes will give an error if they aren't shown in TWS
    contractTuple = ('AAPL', 'STK', 'SMART', 'USD', '', 0.0, '')
    #contractTuple = ('QQQQ', 'OPT', 'SMART', 'USD', '20070921', 47.0, 'CALL')
    #contractTuple = ('ES', 'FUT', 'GLOBEX', 'USD', '200709', 0.0, '')
    #contractTuple = ('ES', 'FOP', 'GLOBEX', 'USD', '20070920', 1460.0, 'CALL')
    #contractTuple = ('EUR', 'CASH', 'IDEALPRO', 'USD', '', 0.0, '')
    for i in range(20):
        stkContract = makeStkContract(contractTuple)
        print ('* * * * REQUESTING MARKET DATA * * * *')
        con.reqMktData(tickId, stkContract, '', False)
        sleep(1)
        print ('* * * * CANCELING MARKET DATA * * * *')
        con.cancelMktData(tickId)
    sleep(1)
    con.disconnect()
    sleep(1)
```

Something that I observed is that if move makeStkContract and cancelMktData, out of loop I get bunch of errors (code 322).

**Comment by ranaroussi** [comment]
I'm a bit baffled as I'm running this code while connected to the demo account, my paper account and my live account. To truly experience and fix this issue, I'd need access to your paper account... 

If you're willing to email me your paper account login details so I can investigate this further, that would help a great deal (you can change your paper account password before and after emailing it to me :)). My email is ran [at] aroussi [dot] com. LMK...

**Comment by rastegar527** [comment]
Thanks Ran. I really appreciate all your help. I'll send you the account information. Please let me know if you can see any data for 
```

rom qtpylib.algo import Also
from qtpylib.indicators import macd#, air
import pandas as pd
from datetime import date time
import datetime

class TestStrategy(Algo):
    def on_start(self):
         pass
    def on_tick(self, instrument):
        pass
    def on_bar(self, instrument):      
        lookback = 28
        for symbol in self.instruments:
            instr = self.get_instrument(symbol)
            bars = instr.get_bars(lookback = loopback)
            print(symbol, len(bars))
        
if __name__ == "__main__":
    strategy = TestStrategy(
        #instruments = prescreening('2014-04-25', '2014-04-29'),
        instruments = ['IBKR', 'AAPL', 'OXY', 'SLB'], 
        resolution  = "1T" # 1Min bar resolution (Pandas "resample" resolutions)
    )
    strategy.run()
```

**Comment by ranaroussi** [comment]
So I logged into your paper account and all seem to work just fine with the latest version `1.5.53`. This is the strategy I've tested:

```python
from qtpylib.algo import Algo
import pandas as pd

class TestStrategy(Algo):
    def on_start(self):
        self.bar_count = dict.fromkeys(self.symbols, 0)
        print("Started...")

    def on_bar(self, instrument):
        bars = instrument.get_bars(28)
        self.bar_count[instrument] = len(bars.index)

        dt = pd.to_datetime(str(bars.index.values[-1])).strftime("%H:%M:%S")
        print(dt, "+1 {:<5}".format(instrument), self.bar_count)

if __name__ == "__main__":
    strategy = TestStrategy(
        bar_window = 28,
        instruments = ['IBKR', 'AAPL', 'OXY', 'SLB'],
        resolution  = "1T"
    )
    strategy.run()
```

Here's a screencast of the above strategy with a 1sec bars setting:
http://screencast-o-matic.com/watch/cDl32JQuXC

...and here's an image of the 1min version (didn't think you'd like to watch a 28min video :))

![min](https://cloud.githubusercontent.com/assets/1185458/21430324/6bcfaba2-c86b-11e6-8b13-d9e8c50f5723.jpg)

**Comment by rastegar527** [comment]
Thanks, Ran. I updated to the last version of the module and I see the data with your code and my paper trading account. However, I still don't see the data with my code I sent you in email. I'm really confused as I don't see any problem with my code. The only difference between your code and my code is the loop for the symbols in on_bar.

**Comment by rastegar527** [comment]
One thing that I observed after running the code for long time is that there are long lag between different symbols data. For example, the bar length for OXY is 28 but at the same time the bar length for IBRK is 7. Is this a connection issue in your opinion or IB server issue? I saw the same issue when I  watch your demo: http://screencast-o-matic.com/watch/cDl32JQuXC

Cheers!

**Comment by ranaroussi** [comment]
That's not actually a delay... Different symbols are receiving data at different times, so they have less bars. If you look at this image, you'll see that, for example, SLB haven't received a single tick yet, while as AAPL received 9, IBKR received 8 and OXY received 7.

<img width="435" alt="screen shot 2016-12-22 at 19 14 42" src="https://cloud.githubusercontent.com/assets/1185458/21434000/ec498b68-c87a-11e6-94a4-1e8a6b154622.png">

AAPL isn't ahead, its just that it has one bar going back... Once all the bars reached `bar_window`, they should all have the same # of bars. I could have deleted bars from faster-moving symbols form self.bars but I voted against it in the meeting I had with myself 😁

---

As for your original code, it runs without problem. I just optimized the code a bit to have a nicer print.

Here's your `on_bar()`, followed by its output:

```python
def on_bar(self, instrument):
    lookback = 28
    for symbol in self.instruments:
        inst = self.get_instrument(symbol)
        bars = inst.get_bars(lookback=lookback)
        print("{:<5}".format(symbol), len(bars))
```

<img width="373" alt="screen shot 2016-12-22 at 19 24 14" src="https://cloud.githubusercontent.com/assets/1185458/21434276/4436c650-c87c-11e6-95c4-77a778c76d1f.png">
